make/targets/openshift/golangci-lint.mk

```diff
@@ -0,0 +1,23 @@
+GOLANGCI_LINT_VERSION ?=1.42.1
+
+GOLANGCI_LINT ?= $(shell which golangci-lint 2>/dev/null)
+
+.ensure-golangci-lint:
+ifeq "" "$(GOLANGCI_LINT)"
+	go get github.com/golangci/golangci-lint/cmd/golangci-lint@v$(GOLANGCI_LINT_VERSION)
```

Comment:
I think is better to use "go install", go get before 1.17 modifies the go.mod IIRC

Reply:
We had some internal discussion with @sttts about this. I'll replace it with curl-based download (similar to what `controller-gen` does).

Reply:
kubernetes use a stub go module for that https://github.com/kubernetes/kubernetes/tree/master/hack/tools
so installing it is like this
https://github.com/kubernetes/kubernetes/blob/master/hack/verify-golangci-lint.sh#L38-L42

Reply:
I copy-pasted bits from `controller-gen`. I'm basically downloading the binaries using CURL or take the one that is already there. I hope it will be fine.

I also check if vendoring Golangci-linter in would be an option but it pull a looooooot of dependencies. So let's try to avoid this during the first pass.

Reply:
nah, just giving alternatives, whatever you think is better